src/tests/functional/test_deploy.py

```diff
@@ -30,7 +30,8 @@
 async def test_weechat_deploy(model, series):
```

Comment:
You are well within your rights to say "That is out of scope for this change" but have you considered updating to the new template so you can verify this change upgrades from a deployed charm? I only mention it because I know we both run this now and I'd hate to see an upgrade failure :)